### PR TITLE
Fix view only persistance

### PR DESCRIPTION
**Pull request by matejkriz**
## Description

Fixes and improvements that I have come across while researching https://github.com/trezor/trezor-suite/issues/12454
I believe they make sense on their own. It's more deletions than additions 🎉

### CODE REVIEW

File: suite-native/graph/src/graphMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import { removeAccountGraphTimeframe } from './slice';
+
+export const graphMiddleware = createMiddleware((action, { next, dispatch }) => {
+    if (deviceActions.forgetDevice.match(action)) {
+        const deviceState = action.payload.state;
+        if (deviceState) {
+            dispatch(removeAccountGraphTimeframe({ deviceState }));
+        }
+    }
```
Comment (juriczech) [inline suite-native/graph/src/graphMiddleware.ts]: Does this have to be middleware? You could just use this action in the reducer and change state

File: suite-common/wallet-core/src/device/deviceReducer.ts [hunk 2]
```diff
@@ -784,23 +784,16 @@ export const selectDeviceFirmwareVersion = memoize((state: DeviceRootState) => {
...
 
     return devices.filter(device => device.id !== PORTFOLIO_TRACKER_DEVICE_ID);
 });
 
 export const selectIsNoPhysicalDeviceConnected = (state: DeviceRootState) => {
-    const devices = selectDevices(state);
+    const devices = selectPhysicalDevices(state);
 
-    return devices.length === 1 && devices[0].id === PORTFOLIO_TRACKER_DEVICE_ID;
+    return !devices.filter(device => device.connected).length;
```
Comment (juriczech) [inline suite-common/wallet-core/src/device/deviceReducer.ts]: For me Array.some would be better for readability, but it's just a nit

File: suite-native/graph/src/graphMiddleware.ts (hunk 1 above)
Comment (matejkriz) [reply suite-native/graph/src/graphMiddleware.ts]: great point! 6827addbe681dc197ce7761978dfd5b8249cf20b

File: suite-common/wallet-core/src/device/deviceReducer.ts (hunk 2 above)
Comment (matejkriz) [reply suite-common/wallet-core/src/device/deviceReducer.ts]: 8a8749319f375d0330bf6d462b88755f13445153

Comment (juriczech) [reply suite-common/wallet-core/src/device/deviceReducer.ts]: Even better 👏

**Review by juriczech — APPROVED** [review]

**Pull request merged**